ocaml/lambda/matching.ml

```diff
@@ -1628,11 +1646,17 @@ and precompile_or ~arg ~arg_sort (cls : Simple.clause list) ors args def k =
               }
             in
             let pm_fv = pm_free_variables orpm in
+            let filter_pm_fv =
```

Comment:
This wants a comment. This should be thought about in tandem with the below comment "variables bound in the or-pattern...".

Reply:
I almost wonder if, instead of creating a function value here, we should instead bind `Typedtree.pat_bound_idents_full ...` to a variable and create `patbound_action_vars` based on that.

Reply:
Yeah, I think that helps clarity. Let me know if you think the new code needs any additional comments.